lib/trento/activity_logging/correlations.ex

```diff
@@ -1,41 +1,20 @@
 defmodule Trento.ActivityLog.Correlations do
   @moduledoc false
 
-  @cache_name :activity_correlations
-  @api_key_regen "api_key_regen"
-  @default_ttl 15_000
+  @callback correlation_key(ctx :: atom()) :: binary()
+  @callback get_correlation_id(key :: binary()) :: binary() | nil
+  @callback put_correlation_id(key :: binary(), value :: binary()) :: :ok
+  @callback expire_correlation_id(key :: binary(), ttl :: integer()) :: :ok
+  def correlation_key(ctx), do: impl().correlation_key(ctx)
+  def get_correlation_id(key), do: impl().get_correlation_id(key)
+  def put_correlation_id(key, value), do: impl().put_correlation_id(key, value)
+  def expire_correlation_id(key, ttl), do: impl().expire_correlation_id(key, ttl)
 
-  def correlation_key(:api_key) do
-    # Allow fetching the correlation key provisioned from the calling
-    # process to prevent leakge of the key across test process boundaries.
-    case Process.get(:correlation_key) do
-      nil ->
-        # non-test scenario, where we want the key to be static
-        @api_key_regen
-
-      key when is_binary(key) ->
-        # test scenario, where we want the key to be owned per-test process
-        key
-    end
-  end
-
-  @spec put_correlation_id(binary(), binary()) :: :ok
-  def put_correlation_id(key, value) do
-    _ = Cachex.put(@cache_name, key, value)
-    :ok
-  end
-
-  @spec get_correlation_id(binary()) :: binary() | nil
-  def get_correlation_id(key) do
-    with {:ok, maybe_value} <- Cachex.get(@cache_name, key) do
-      maybe_value
-    end
-  end
-
-  @spec expire_correlation_id(binary(), non_neg_integer()) :: :ok
-  def expire_correlation_id(key, ttl \\ @default_ttl) do
-    # ttl unit is milliseconds
-    _ = Cachex.expire(@cache_name, key, ttl)
-    :ok
-  end
+  defp impl,
+    do:
+      Application.get_env(
+        :trento,
+        :correlations,
+        Trento.ActivityLog.Correlations.UnscopedCorrelations
```

Comment:
I know it does not change much, however for consistency I'd move this to the config.

Something like
```elixir
config :trento, correlations: Trento.ActivityLog.Correlations.UnscopedCorrelations
```